### ISSUE TITLE
pieces going right way

### ISSUE THREAD

**Issue by amlwwalker** [nielslange/react-backgammon#1]
Hi,
Nice work in setting this board up - however am I right to think that the pieces for each player are going the wrong way around the board from the initial set up?
I uncommented the start arrangement in your state.tsx and i think they are going the wrong way around?

**Comment by nielslange** [comment]
Thanks for reaching out, @amlwwalker. You are absolutely right. Recently, I refactored the codebase and did not notice that I broke the direction of the checkers. I just corrected this. Thanks again for bringing this issue to my attention, and if you spot anything else, just let me know. That said, the game is not 100% ready yet, so it's possible that a few things are not working as expected yet.

**Comment by amlwwalker** [comment]
Hi @nielslange 
Good to know I wasn't going mad 👍 
What is your objective with this? Would you be interested in any collaboration? I worked on a backgammon education app at one point but chose a terrible UI and the pains of that slowed my development down. 

I have a few questions:

- How easy is it to specify layouts (start positions etc)?
- How easy is it to define what rules to use? I see that the rules are _nearly_ there in the game as it stands. However things like being able to flip player and not being able to choose which dice to move first are not quite there for a set of rules yet....?

I ask because there are other games played on a backgammon board with different start positions and different rules. I was wondering if you would be interested in a slight refactor and potentially working together to get this to a place where it can accomodate providing different starts/rules and building in some education potentially? One thing for instance would be to play out games based on a JSON object of the dice/pieces etc and where they should move to.

I ask, because your app is by far the best looking and closest I have seen to a "nearly ready to build on" backgammon app and it would be sweet to put it to good use


Starting small ... thoughts?


P.S just playing with  the app a bit more, i don't think you can take players currently?

**Comment by nielslange** [comment]
Hello @amlwwalker 👋

Sorry for my delayed reply. 

> What is your objective with this? Would you be interested in any collaboration? I worked on a backgammon education app at one point but chose a terrible UI and the pains of that slowed my development down.

I'd be very much interested in a collaboration. I started this project as I love playing both backgammon and [tric trac](https://en.wikipedia.org/wiki/Trictrac), a variant of backgammon. As I could not find an app to play tric trac, I decided to develop my own game. 

> How easy is it to specify layouts (start positions etc)?

That's pretty easy to implement as the start positions are defined in this file via a Redux data store: https://github.com/nielslange/react-backgammon/blob/trunk/src/data/state.ts I could easily outsource that state into JSON files, to load different start positions.

> How easy is it to define what rules to use? I see that the rules are nearly there in the game as it stands. However things like being able to flip player and not being able to choose which dice to move first are not quite there for a set of rules yet....?

I do struggle a bit with finding the best pattern to apply rules. I tried different approaches, but I wasn't happy with any of them. That said, if you have a good idea, I'm all ears. 

> I ask, because your app is by far the best looking and closest I have seen to a "nearly ready to build on" backgammon app and it would be sweet to put it to good use

That's great to hear. 😊 

> P.S just playing with the app a bit more, i don't think you can take players currently?

Currently, the players automatically switch once one player made his moves. However, I'd love to add functionality to be able:
- to play against a bot, maybe AI powered
- to play against other players online

Long story short, I'm more than open to collaborating with you on this project. 😀

**Comment by amlwwalker** [comment]
emailing you now @nielslange

**Comment by amlwwalker** [comment]
hey @nielslange  - just want to check that you received an email from me

**Comment by nielslange** [comment]
I did, @amlwwalker. Replying now. 🚀
